### PR TITLE
Set custom player name, cl_fakenick = 0 to only use Steam name, postfix index duplicate names

### PR DESCRIPTION
**Pull request by nullsystem**
* `neo_name` - To set a custom name so it can be used instead of Steam name
* `cl_fakenick` - Set to 1 to see others + your own `neo_name` if those are set, otherwise only view steam name
* Duplicate checker - Due to putting in custom `neo_name`, now roll our own duplicate checker which postfix ` (%d)` if the name is duplicate indexed
* `neo_name` been checked to been applied for scoreboard, event indicator, killfeed, damage info, friendly markers. Other places either may also inherits it if they use `g_PR->GetPlayerName`/similar.

### CODE REVIEW

**Comment by AdamTadeusz** [comment]
Hud stuff made the player squad variable networked, conflicting with the duped name stuff

**Comment by nullsystem** [comment]
@AdamTadeusz Resolved the conflicts and made damage info use neo_name + duplicator indexes now.

File: mp/src/game/client/c_playerresource.cpp [hunk 1]
```diff
@@ -28,6 +28,8 @@ IMPLEMENT_CLIENTCLASS_DT_NOBASE(C_PlayerResource, DT_PlayerResource, CPlayerReso
 #ifdef NEO
 	RecvPropArray3(RECVINFO_ARRAY(m_iXP), RecvPropInt(RECVINFO(m_iXP[0]))),
 	RecvPropArray3(RECVINFO_ARRAY(m_iClass), RecvPropInt(RECVINFO(m_iClass[0]))),
+	RecvPropArray3(RECVINFO_ARRAY(m_szNeoName), RecvPropString(RECVINFO(m_szNeoName[0]))),
+	RecvPropArray3(RECVINFO_ARRAY(m_szNeoNameDupeIdx), RecvPropInt(RECVINFO(m_szNeoNameDupeIdx[0]))),
```
Comment (AdamTadeusz) [inline mp/src/game/client/c_playerresource.cpp:32]: if this is an int should it be called m_iNeoNameDupeIdx?

**Review by AdamTadeusz — DISMISSED** [review]

File: mp/src/game/shared/neo/neo_gamerules.cpp [hunk 2]
```diff
@@ -1307,6 +1310,90 @@ void CNEORules::ClientSettingsChanged(CBasePlayer *pPlayer)
 	}
 	pNEOPlayer->SetDefaultFOV(pNEOPlayer->ClientFOV());
 
+	const char *pszSteamName = engine->GetClientConVarValue(pPlayer->entindex(), "name");
+
+	const bool clientAllowsNeoName = (1 == atoi(engine->GetClientConVarValue(engine->IndexOfEdict(pNEOPlayer->edict()), "cl_fakenick")));
```
[the remark this answers is not in the document]
Comment (Rainyan) [reply mp/src/game/shared/neo/neo_gamerules.cpp:1315]: > With the new changes can we use atoi again?

Itoa is nonstandard, but I think we should have atoi available on all platforms(?)

**Review by AdamTadeusz — APPROVED** [review]

File: mp/src/game/shared/neo/neo_gamerules.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (nullsystem) [reply mp/src/game/shared/neo/neo_gamerules.cpp:1315]: https://en.cppreference.com/w/cpp/17

Due to being limited to GCC 10 support, `std::to/from_chars` is only good for integers, but not floating-pointer numbers, that comes in GCC 11.

**Pull request merged**